# **Data Collection Notebook**

## Objectives

* To fetch the data from Kaggle and save it as raw data.
* Inspect the data and save it under: outputs/datasets/collection

## Inputs

* Kaggle JSON File - this is the authentication token to grant access to the dataset.

## Outputs

* To generate the Datasets to:
    
    * outputs/datasets/collection/HousePriceRecords.csv
    * outputs/datasets/collection/InheritedHouses.csv

## Additional Comments

* The HousePriceRecords.csv is the data to be used to build the machine learning models.
* The InheritedHouses.csv is the data with the inherited houses whose prices the client would like to predict. 


---

# Change working directory

* We need to change the working directory from its current folder to its parent folder

We need to change the working directory from its current folder to its parent folder
* We access the current directory with os.getcwd()

In [17]:
import os
current_dir = os.getcwd()
current_dir

'/workspace/hertiage-housing'

We want to make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chir() defines the new current directory

In [18]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


Confirm the new current directory

In [19]:
current_dir = os.getcwd()
current_dir

'/workspace'

---

# Fetch the data from Kaggle

"**First we must install the Kaggle package in order to fetch the data:**"

In [20]:
%pip install kaggle==1.5.12

Note: you may need to restart the kernel to use updated packages.


"**If it is already installed, this step can be skipped.**"

"Next we will use the **kaggle.json** token to be able to authenticate and access the data:"

In [21]:
import os
os.environ['KAGGLE_CONFIG_DIR'] = os.getcwd()
! chmod 600 kaggle.json

chmod: cannot access 'kaggle.json': No such file or directory


The client has provided the following dataset: [Housing Prices](/www.kaggle.com/datasets/codeinstitute/housing-prices-data)

Firstly, the *Dataset* needs to be downloaded and set to a destination folder:

In [22]:
KaggleDatasetPath = "codeinstitute/housing-prices-data"
DestinationFolder = "inputs/datasets/raw"   
! kaggle datasets download -d {KaggleDatasetPath} -p {DestinationFolder}

Traceback (most recent call last):
  File "/workspace/hertiage-housing/.venv/bin/kaggle", line 5, in <module>
    from kaggle.cli import main
  File "/workspace/hertiage-housing/.venv/lib/python3.12/site-packages/kaggle/__init__.py", line 23, in <module>
    api.authenticate()
  File "/workspace/hertiage-housing/.venv/lib/python3.12/site-packages/kaggle/api/kaggle_api_extended.py", line 164, in authenticate
    raise IOError('Could not find {}. Make sure it\'s located in'
OSError: Could not find kaggle.json. Make sure it's located in /workspace. Or use the environment method.


Finally, the downloaded file needs to be unzipped, and the **kaggle.json** file deleted:

In [23]:
! unzip {DestinationFolder}/*.zip -d {DestinationFolder} \
  && rm {DestinationFolder}/*.zip \
  && rm kaggle.jso

unzip:  cannot find or open inputs/datasets/raw/*.zip, inputs/datasets/raw/*.zip.zip or inputs/datasets/raw/*.zip.ZIP.

No zipfiles found.


---

# Load and Inspect House Price Records

Next we will load and inspect the **House Price Records** data:

In [29]:
import pandas as pd
df_house_prices = pd.read_csv(f"inputs/datasets/raw/house-price-20211124T154130Z-001/house-price/house_prices_records.csv")
print(df_house_prices.shape)
df_house_prices.head()

FileNotFoundError: [Errno 2] No such file or directory: 'inputs/datasets/raw/house-price-20211124T154130Z-001/house-price/house_prices_records.csv'

Next we will look at the *Dataframe* summary:

In [10]:
df_house_prices.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 24 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   1stFlrSF       1460 non-null   int64  
 1   2ndFlrSF       1374 non-null   float64
 2   BedroomAbvGr   1361 non-null   float64
 3   BsmtExposure   1460 non-null   object 
 4   BsmtFinSF1     1460 non-null   int64  
 5   BsmtFinType1   1346 non-null   object 
 6   BsmtUnfSF      1460 non-null   int64  
 7   EnclosedPorch  136 non-null    float64
 8   GarageArea     1460 non-null   int64  
 9   GarageFinish   1298 non-null   object 
 10  GarageYrBlt    1379 non-null   float64
 11  GrLivArea      1460 non-null   int64  
 12  KitchenQual    1460 non-null   object 
 13  LotArea        1460 non-null   int64  
 14  LotFrontage    1201 non-null   float64
 15  MasVnrArea     1452 non-null   float64
 16  OpenPorchSF    1460 non-null   int64  
 17  OverallCond    1460 non-null   int64  
 18  OverallQ

Next we will check for duplicated data:

In [11]:
df_house_prices[df_house_prices.duplicated(subset=None)]

Empty DataFrame
Columns: [1stFlrSF, 2ndFlrSF, BedroomAbvGr, BsmtExposure, BsmtFinSF1, BsmtFinType1, BsmtUnfSF, EnclosedPorch, GarageArea, GarageFinish, GarageYrBlt, GrLivArea, KitchenQual, LotArea, LotFrontage, MasVnrArea, OpenPorchSF, OverallCond, OverallQual, TotalBsmtSF, WoodDeckSF, YearBuilt, YearRemodAdd, SalePrice]
Index: []

[0 rows x 24 columns]

There is no duplicated data.

#### Findings:

The *Dataset* contains **1460** entries of data and **24** columns.

here is a mix of the follow data types: <code>int64</code>,<code>float64</code> and <code>object</code>.

---

# Load and Inspect Inherited Houses

Next we will load and inspect the **Inherited Houses** data:

In [12]:
df_inherited_houses = pd.read_csv(f"inputs/datasets/raw/house-price-20211124T154130Z-001/house-price/inherited_houses.csv")
print(df_inherited_houses.shape)
df_inherited_houses.head()

(4, 23)


1stFlrSF  2ndFlrSF  BedroomAbvGr BsmtExposure  BsmtFinSF1 BsmtFinType1  \
0       896         0             2           No       468.0          Rec   
1      1329         0             3           No       923.0          ALQ   
2       928       701             3           No       791.0          GLQ   
3       926       678             3           No       602.0          GLQ   

   BsmtUnfSF  EnclosedPorch  GarageArea GarageFinish  ...  LotArea  \
0      270.0              0       730.0          Unf  ...    11622   
1      406.0              0       312.0          Unf  ...    14267   
2      137.0              0       482.0          Fin  ...    13830   
3      324.0              0       470.0          Fin  ...     9978   

   LotFrontage MasVnrArea  OpenPorchSF  OverallCond  OverallQual  TotalBsmtSF  \
0         80.0        0.0            0            6            5        882.0   
1         81.0      108.0           36            6            6       1329.0   
2         74.0        0.0           34            5            5        928.0   
3         78.0       20.0           36            6            6        926.0   

   WoodDeckSF  YearBuilt  YearRemodAdd  
0         140       1961          1961  
1         393       1958          1958  
2         212       1997          1998  
3         360       1998          1998  

[4 rows x 23 columns]

Next we will look at the *Dataframe* summary:

In [13]:
df_inherited_houses.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4 entries, 0 to 3
Data columns (total 23 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   1stFlrSF       4 non-null      int64  
 1   2ndFlrSF       4 non-null      int64  
 2   BedroomAbvGr   4 non-null      int64  
 3   BsmtExposure   4 non-null      object 
 4   BsmtFinSF1     4 non-null      float64
 5   BsmtFinType1   4 non-null      object 
 6   BsmtUnfSF      4 non-null      float64
 7   EnclosedPorch  4 non-null      int64  
 8   GarageArea     4 non-null      float64
 9   GarageFinish   4 non-null      object 
 10  GarageYrBlt    4 non-null      float64
 11  GrLivArea      4 non-null      int64  
 12  KitchenQual    4 non-null      object 
 13  LotArea        4 non-null      int64  
 14  LotFrontage    4 non-null      float64
 15  MasVnrArea     4 non-null      float64
 16  OpenPorchSF    4 non-null      int64  
 17  OverallCond    4 non-null      int64  
 18  OverallQual   

Next we will check for duplicated data:

In [14]:
df_inherited_houses[df_inherited_houses.duplicated(subset=None)]

Empty DataFrame
Columns: [1stFlrSF, 2ndFlrSF, BedroomAbvGr, BsmtExposure, BsmtFinSF1, BsmtFinType1, BsmtUnfSF, EnclosedPorch, GarageArea, GarageFinish, GarageYrBlt, GrLivArea, KitchenQual, LotArea, LotFrontage, MasVnrArea, OpenPorchSF, OverallCond, OverallQual, TotalBsmtSF, WoodDeckSF, YearBuilt, YearRemodAdd]
Index: []

[0 rows x 23 columns]

There is no duplicated data.

#### Findings:

* The *Dataset* contains **4** entries of data and **23** columns.
* There is a mix of the follow data types: <code>int64</code>,<code>float64</code> and <code>object</code>.

---

# Push files to Repo

* If you do not need to push files to Repo, you may replace this section with \"Conclusions and Next Steps\" and state your conclusions and next steps.

In [16]:
import os
try:
  os.makedirs(name='outputs/datasets/collection')
except Exception as e:
  print(e)

df_house_prices.to_csv(f"outputs/datasets/collection/HousePriceRecords.csv", index=False)
df_inherited_houses.to_csv(f"outputs/datasets/collection/InheritedHouses.csv", index=False)